### PR TITLE
docs: update latest release v0.9.1

### PR DESCRIPTION
**Pull request by v0y4g3r**
## What's Changed in this PR

Add release doc for v0.9.1

*Describe the change in this PR*

## Checklist

- [x] Please confirm that all corresponding versions of the documents have been revised.
- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/release-notes/release-0-9-1.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# v0.9.1
+
+Release date: August 02, 2024
```
Comment (killme2008) [inline docs/release-notes/release-0-9-1.md:3]: Missing highlights

Comment (v0y4g3r) [reply docs/release-notes/release-0-9-1.md:3]: > Missing highlights or main changes summary.

fixed in #[2603ec9](https://github.com/GreptimeTeam/docs/pull/1120/commits/2603ec9de98073357f2a134e0cd9849e18f17236). Let me also update the release SOP doc.

File: docs/release-notes/release-0-9-1.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
```
Comment (nicecui) [inline docs/release-notes/release-0-9-1.md:4]:
```suggestion
## 👍 Highlights
```

Comment (v0y4g3r) [reply docs/release-notes/release-0-9-1.md:4]: > 

It's not actually *highlights*. I folllowed the style here https://github.com/GreptimeTeam/docs/blob/2603ec9de98073357f2a134e0cd9849e18f17236/docs/release-notes/release-0-8-1.md?plain=1#L5-L7

**Review by nicecui — DISMISSED** [review]

**Review by nicecui — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]

**Pull request merged**